pandas/core/tools/datetimes.py

```diff
@@ -182,9 +183,12 @@ def _guess_datetime_format_for_array(arr, **kwargs):
         return _guess_datetime_format(arr[non_nan_elements[0]], **kwargs)
 
 
+@deprecate_kwarg(old_arg_name='utc', new_arg_name='tz',
```

Comment:
instead of doing this, simply translate ``utc=True`` to ``tz='UTC'`` in the code itself (and issue a deprecation warning). Further you will need to raise if ``utc=True`` and ``tz is not None``, so need a bit more handling (and add a test for this).